packages/babel-generator/src/printer.js

```diff
@@ -291,18 +292,24 @@ export default class Printer {
    *  `undefined` will be returned and not `foo` due to the terminator.
    */
 
-  startTerminatorless(): Object {
-    return (this._parenPushNewlineState = {
-      printed: false,
-    });
+  startTerminatorless(key: string): Object {
+    if (key == "label") {
```

Comment:
This still seems like an implementation detail to me, better to do

```
startTerminatorless(isLabel: boolean = false): Object {
```
with the `key` check staying in the calling function.